### PR TITLE
chore: use base images from mcr

### PR DESCRIPTION
**Pull request by aramase**
- use base images from mcr
- bump distroless-iptables to `v0.4.3`

### CODE REVIEW

File: docker/webhook.Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-ARG BUILDER=mcr.microsoft.com/oss/go/microsoft/golang:1.21-bullseye
```
Comment (sozercan) [inline docker/webhook.Dockerfile:1]: are we using these as args anywhere? if not, let's add them directly and have dependabot manage these (applies to all)

Comment (aramase) [reply docker/webhook.Dockerfile:1]: done!

File: docker/proxy.Dockerfile [hunk 2]
```diff
@@ -1,7 +1,6 @@
-ARG BUILDER=mcr.microsoft.com/oss/go/microsoft/golang:1.21-bullseye
-ARG BASEIMAGE=gcr.io/distroless/static:nonroot
+ARG BASEIMAGE=mcr.microsoft.com/cbl-mariner/distroless/minimal:2.0-nonroot
```
Comment (sozercan) [inline docker/proxy.Dockerfile:1]: let's add it for this too so dependabot can pin to digest

File: examples/msal-python/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ARG BUILDER=debian:11-slim
-ARG BASEIMAGE=gcr.io/distroless/python3-debian11
+ARG BASEIMAGE=mcr.microsoft.com/cbl-mariner/distroless/python:3.9
```
Comment (sozercan) [inline examples/msal-python/Dockerfile:2]: same here

File: examples/msal-python/Dockerfile [hunk 4]
```diff
@@ -1,8 +1,4 @@
-ARG BUILDER=debian:11-slim
-ARG BASEIMAGE=gcr.io/distroless/python3-debian11
-
-# ref: https://github.com/GoogleContainerTools/distroless/blob/main/examples/python3-requirements/Dockerfile
-FROM ${BUILDER}  AS build
+FROM debian:11-slim  AS build
```
Comment (sozercan) [inline examples/msal-python/Dockerfile:1]:
```suggestion
FROM mcr.microsoft.com/mirror/docker/library/debian:bookworm-slim AS build
```

**Review by sozercan — APPROVED** [review]

**Pull request merged**